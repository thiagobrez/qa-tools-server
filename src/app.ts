import express from 'express';
import cors from 'cors';
import { connect } from 'mongoose';
import routes from './routes';

class App {
  public express: express.Application;

  public constructor() {
    this.express = express();

    this.middlewares();
    this.database();
    this.routes();
  }

  private middlewares(): void {
    this.express.use(express.json());
    this.express.use(cors());
  }

  private database(): void {
    connect(
      'mongodb://localhost:27017/qa-tools',
      { useNewUrlParser: true },
    );
  }

  private routes(): void {
    this.express.use(routes);
  }
}

export default new App().express;
