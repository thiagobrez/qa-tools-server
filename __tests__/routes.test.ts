import { connection } from 'mongoose';
import app from '../src/app';

const request = require('supertest');

afterAll(async () => {
  await connection.close();
});

it('reads all users', async () => {
  const res = await request(app).get('/users');
  const thiago = {
    _id: '5d38aa3ec832580ef81c3ed7',
    email: 'thiago@email.com',
    firstName: 'Thiago',
    lastName: 'Brezinski',
    createdAt: '2019-07-24T18:58:06.350Z',
    updatedAt: '2019-07-24T18:58:06.350Z',
    __v: 0,
  };

  expect(res.status).toBe(200);
  expect(res.body).toBeInstanceOf(Array);
  expect(res.body).toContainEqual(thiago);
  expect(res.body).toMatchSnapshot();
});
