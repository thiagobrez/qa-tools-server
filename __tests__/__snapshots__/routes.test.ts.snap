// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reads all users 1`] = `
Array [
  Object {
    "__v": 0,
    "_id": "5d38aa3ec832580ef81c3ed7",
    "createdAt": "2019-07-24T18:58:06.350Z",
    "email": "thiago@email.com",
    "firstName": "Thiago",
    "lastName": "Brezinski",
    "updatedAt": "2019-07-24T18:58:06.350Z",
  },
]
`;
